Add WEBP + AVIF Support by default

in-session opened the pull request:
Webp should already be set up by default

### Steps to reproduce:
1. Add webp, avif, etc files to lib/web/images directory.
2. Modify app/code/Magento/Backend/view/adminhtml/layout/admin_login.xml file to use webp, avif, etc file as Magento 2 logo (edit "logo" block "logo_image_src" argument value).
3. Flush cache.
4. Check webp, avif, etc file is used as Magento 2 logo on the admin login page.


### Resolved issues:
1. [x] resolves magento/magento2#32495: Add WEBP + AVIF Support by default

in-session committed:
Add WEBP Support by default

in-session committed:
Add WEBP Support by default

ihor-sviziev reviewed (changes requested):
Hi @in-session,
Could you sign Adobe CLA and update the nginx config sample as well?
https://github.com/magento/magento2/blob/2.4-develop/nginx.conf.sample

in-session committed:
Add WEBP Support by default

ihor-sviziev commented:
Yeah, make sense for sure

On Tue, 9 Mar 2021 at 13:30, Pieter Hoste <notifications@github.com> wrote:

> Maybe add the new avif image format as well? It's not supported by many
> browsers yet <https://caniuse.com/avif>, but will be in the future ...
>
> Some extension developers are already starting to implement it:
>
>    - https://www.yireo.com/blog/2020-12-29-magento2-webp-becomes-next-gen
>    -
>    https://www.jajuma.de/en/jajuma-develop/extensions/ultimate-image-optimizer-extension-for-magento-2
>    - ...
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/magento/magento2/pull/32400#issuecomment-793751206>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAOJOUPSC3FMNIV6FVXL67DTCYBETANCNFSM4Y3JOOMA>
> .
>

in-session committed:
Update nginx.conf.sample

in-session committed:
Update .htaccess

in-session committed:
Update .htaccess

ihor-sviziev commented:
@magento run all tests

hostep commented:
Maybe also avif sequences? https://github.com/h5bp/server-configs-apache/pull/251

https://github.com/h5bp/server-configs-apache/ is my go to project for verifying apache config settings, maybe double check webp as well? (sorry lack of time here for proper review)

in-session commented:
Ok let me check i will add also some other like opentype fonts

in-session committed:
Update nginx.conf.sample

in-session committed:
Update .htaccess

in-session committed:
Update .htaccess

in-session committed:
Update .htaccess

in-session commented:
> Maybe also avif sequences? [h5bp/server-configs-apache#251](https://github.com/h5bp/server-configs-apache/pull/251)
> 
> https://github.com/h5bp/server-configs-apache/ is my go to project for verifying apache config settings, maybe double check webp as well? (sorry lack of time here for proper review)

Have you some information about nginx?
`location ~* \.(ico|jpg|jpeg|png|gif|svg|webp|avif|avifs`

or

`location ~* \.(ico|jpg|jpeg|png|gif|svg|webp|avif|avifs?`

ihor-sviziev commented:
@in-session I believe it's better to use `location ~* \.(ico|jpg|jpeg|png|gif|svg|webp|avif|avifs`  as it will be just simpler to read

in-session committed:
Update .htaccess

in-session committed:
Update .htaccess

in-session committed:
Update .htaccess

in-session committed:
Update .htaccess

in-session committed:
Remove Flash

in-session committed:
Remove Flash

in-session committed:
Update .htaccess

in-session committed:
Update .htaccess

in-session committed:
Remove Flash

in-session committed:
Add webmanifest for Static files

in-session committed:
SVGZ Support

in-session committed:
Add svgz support

in-session committed:
Update nginx.conf.sample

in-session committed:
Update .htaccess

in-session commented:
Ok I think that is good for now

Added Media + Static
- svgz
- webp
- avif
- avifs

Removed Media + Static
- swf

Added Static
- webmanifest
- font/truetype
- font/opentype

Maybe someone can go through the nginx.conf.sample again?

ihor-sviziev commented on pub/static/.htaccess:
```diff
@@ -111,14 +122,17 @@ AddType application/xml xml
 
     # Favicon, images, flash
-    <FilesMatch \.(ico|gif|png|jpg|jpeg|swf|svg)$>
+    <FilesMatch \.(ico|gif|png|jpg|jpeg|svg|svgz|webp|avif|avifs?)$>
```
I believe the `?` at the end is useless in this case as you listed both `avif` and `avifs` in the list:
```suggestion
    <FilesMatch \.(ico|gif|png|jpg|jpeg|svg|svgz|webp|avif|avifs)$>
```

ihor-sviziev commented on pub/static/.htaccess:
```diff
@@ -72,7 +77,7 @@ AddType application/xml xml
 <IfModule mod_headers.c>
 
-    <FilesMatch .*\.(ico|jpg|jpeg|png|gif|svg|js|css|swf|eot|ttf|otf|woff|woff2|html|json)$>
+    <FilesMatch .*\.(ico|jpg|jpeg|png|gif|svg|webp|avif|avifs?|js|css|eot|ttf|otf|woff|woff2|html|json|webmanifest)$>
```
```suggestion
    <FilesMatch .*\.(ico|jpg|jpeg|png|gif|svg|webp|avif|avifs|js|css|eot|ttf|otf|woff|woff2|html|json|webmanifest)$>
```

ihor-sviziev reviewed (changes requested):
Could you check my comments and then squash all your commits to single one?

in-session committed:
Update .htaccess

in-session committed:
Update .htaccess

ihor-sviziev reviewed (changes requested):
Could you squash all your changes to single commit?

in-session commented:
> Could you squash all your changes to single commit?
It seems like I can't do that.

in-session committed:
Update .htaccess

ihor-sviziev commented:
@in-session here is a good example of how you can squash all commits: https://www.internalpointers.com/post/squash-commits-into-one-git
Please note that after all your changes, you need to force push your changes.

In case of any questions - please reach me at EngCom Slack.

in-session committed:
Merge branch '2.4-develop' into 2.4-develop

ihor-sviziev commented:
Let's leave it not squashed.
@magento run all tests

ihor-sviziev commented:
@magento run Functional Tests B2B

magento-engcom-team commented:
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8925](https://jira.corp.magento.com/browse/ENGCOM-8925) has been created to process this Pull Request

magento-engcom-team commented:
@in-session thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

ihor-sviziev commented:
@magento run Functional Tests B2B

ihor-sviziev commented:
@magento run Functional Tests B2B

engcom-Oscar commented:
@engcom-Foxtrot can you please check this PR, it require dev-experience.

engcom-Foxtrot commented:
QA passed

engcom-Oscar commented:
@magento create issue

engcom-Foxtrot committed:
Merge branch '2.4-develop' into 2.4-develop

engcom-Foxtrot commented:
@magento run all tests
